mgmtsystem_action/tests/test_create_action.py

```diff
@@ -70,18 +64,17 @@ def test_get_action_url(self):
             '&id={}&model={}'.format(self.record.id, self.record._name), url
         )
 
+    @freeze_time('2019-06-05')
```

Comment:
Please don't introduce libraries like this one only for testing and more without a strong need.

Reply:
it is a strong need because is the only way to test the real feature of sending emails in the future as it would work out of tests.

Also, this lib is used already in other OCA repos https://github.com/search?q=org%3AOCA+freezegun&type=Code

Many times I see hacks to test such cases (like dates passed in ctx only for testing) and that's worse than adding an external lib "just" for testing IMO.

Reply:
It has been used only in your devs, not in whole OCA.

Odoo has included a tool for doing that in tests, but not sure if it's for v13 or v12. Any way, I don't want this hidden dependency breaking my internal CIs as I'm using these modules. The test was previously there and it was working, and it's totally unrelated to the purpose of this PR.

Reply:
I  know the test was there, I coded it ;)
BUT it was failing on this new PR and it was a custom hack I've decided to use a well know library to handle it.
Is not hidden, it's declared in requirements.txt and I bet you check that or include that when you add a new repository to your project. 
If not, then I  wonder how you pick dependencies for new repos :thinking: 
Regarding the v12/13 feature: I don't see anything. Any pointer?

Reply:
We don't aggregate all the requirements.txt of all the used repositories because we don't use every module!

I tweeted some time ago about it, but I'm not able to find it right now, and Twitter search is terrible (well, it doesn't exists at all).

Do the import/test conditionally at least for not being executed if no library present.

Reply:
> We don't aggregate all the requirements.txt of all the used repositories because we don't use every module!

Maybe that's what we should fix?

Reducing the test coverage is probably not the best course of action when the problem is only a missing pip install?

Reply:
I don't want extra burden Python libraries in my image if not used, as there are also other things to perform with some pip installs (for example, requiring packages for sources, or other libraries, like mysql and similar in server-backend repo).

Any way, the main problem here is changing this on an already stable module. This will break my current instances CI with this module deployed.

Reply:
Sorry to get in the middle, but I came across this one because I also use a lot ```freezegun``` to make tests that traverse in time.

@simahawk : If for whathever reason ```freezegun``` is banned on OCA and you still want to contribute this, you can use ```mock``` instead, which should not be a problem because it is listed as part of Odoo's ```requirements.txt```. For instance, Odoo's core (V12) uses it on this test to go back in time: https://github.com/odoo/odoo/blob/e1833936255839cd34eba5758ca4c3124fd2522c/addons/account/tests/test_account_customer_invoice.py#L394. However in my tests I had to patch also ```Datetime.now``` because sometimes internally Odoo uses a datetime, this way:

```python
from mock import patch
# ...
# Then inside the test:
with patch('odoo.fields.Date.today', lambda: datetime.date(2018, 1, 3)),
        patch('odoo.fields.Datetime.now', lambda: datetime.datetime(2018, 1, 3)):
    # The code that tests in the past.
    # ...
```

It's more painful to use than ```freezegun```, but can be an alternative to it.

Reply:
@BT-cserra tnx for the feedback. Yes, I did that sometimes but reality is the - as you said - is a bit painful because it's not granted to be applied to every places whereas a date or time is implied. 
On the contrary, freezegun does exactly what you expect, no matter the place.

Regarding the ban, frankly I don't understand @pedrobaeza ... to me is really not a point.
To me this lib should stay everywhere in OCA and ppl should be encouraged to use it instead of mocking things every time in a different way.

Reply:
I don't ban the library itself. What I ban is to introduce it in a module already landed in stable, making existing deployments with linked CIs to fail.

Reply:
then I wonder how do you update your stable CI?
You don't check what's coming thru? You don't take care of checking what's changed?
And then, even if it breaks your CI, is going to be an easy fix: add the lib to requirements and you are done.
My friend, honestly I think we are just losing time here, sorry :angel:

Reply:
Of course it will fail, but I don't want to change all my scaffolding for introducing that library in them. My reasoning is that you are using it only marginally here, so a workaround like the propose one is good for that. If you would introduce it at the beginning, then it's part of the deployment you need to do when the module is introduced in the instance.

Reply:
And I repeat the title and the purpose of this PR has nothing to do with this change.

Reply:
> Of course it will fail, but I don't want to change all my scaffolding for introducing that library in them. My reasoning is that you are using it only marginally here, so a workaround like the propose one is good for that. If you would introduce it at the beginning, then it's part of the deployment you need to do when the module is introduced in the instance.

So then we have to use workarounds everywhere because you cannot update your scaffolding? Doesn't really make sense to me.

Reply:
> making existing deployments with linked CIs to fail.

Isn't it the purpose of the CI by the way, so you can, for instance detect and add the missing libs? :wink:

Reply:
@guewen there are 3 main reasons for not liking this:

- The PR was not for that. See the title and the purpose of it.
- It's introducing the dependency on a stable branch well established, with several customers that already have this module deployed.
- This is something only for testing, and only in one place, not adding any value to the module itself.